### PR TITLE
Fix: TLS support

### PR DESCRIPTION
**Pull request by jjnicola**
**What**:
Fix: TLS support

**Why**:
The server didn't start if the client certs were missing. With this fix, it is possible to start the server in TLS or mTLS modes

**How**:
Start openvasd without client cert options, without a valid directory or an empty directory.

**Checklist**:

- [ ] Tests
- [ ] PR merge commit message adjusted

### CODE REVIEW

**Review by nichtsfrei — DISMISSED** [review]

**Review by Kraemii — CHANGES_REQUESTED** [review]
Not sure about this fix, as the root cause why there are no client certs is not stated, even if it is clear, that either no path is configured, the given path is empty or the given path does not exist. Also not giving a client certs path could be wanted, when just TLS is sufficient

**Comment by jjnicola** [comment]
> Not sure about this fix, as the root cause why there are no client certs is not stated, even if it is clear, that either no path is configured, the given path is empty or the given path does not exist. Also not giving a client certs path could be wanted, when just TLS is sufficient

Fix: Improved the warnings and clarify why is running in TLS mode if client certs are not found.

File: rust/openvasd/src/tls.rs [hunk 1]
```diff
@@ -175,24 +172,60 @@ pub fn tls_config(config: &crate::config::Config) -> Result<Option<TlsData>, Err
...
-        .with_single_cert(certs, key)?;
+        config.alpn_protocols = vec![b"h2".to_vec()];
 
-    config.alpn_protocols = vec![b"h2".to_vec()];
+        Ok(Some((client_identifier, config, !clients.is_empty())))
+    } else {
+        match &config.tls.client_certs {
+            Some(clicerts) => {
+                match std::fs::read_dir(clicerts) {
+                    Ok(x) => {
```
Comment (Kraemii) [inline rust/openvasd/src/tls.rs:199]:
```suggestion
                    Ok(_) => {
```

Comment (Kraemii) [reply rust/openvasd/src/tls.rs:199]: cargo clippy is complaining

File: rust/openvasd/src/tls.rs [hunk 2]
```diff
@@ -175,24 +172,60 @@ pub fn tls_config(config: &crate::config::Config) -> Result<Option<TlsData>, Err
...
-        .with_single_cert(certs, key)?;
+        config.alpn_protocols = vec![b"h2".to_vec()];
 
-    config.alpn_protocols = vec![b"h2".to_vec()];
+        Ok(Some((client_identifier, config, !clients.is_empty())))
+    } else {
+        match &config.tls.client_certs {
+            Some(clicerts) => {
+                match std::fs::read_dir(clicerts) {
+                    Ok(_x) => {
```
Comment (Kraemii) [inline rust/openvasd/src/tls.rs:199]:
```suggestion
                    Ok(_) => {
```
I would rather remove the x completely, or is there a reason for this?

Comment (jjnicola) [reply rust/openvasd/src/tls.rs]: it was the clippy's suggestion.

**Review by Kraemii — APPROVED** [review]

File: rust/openvasd/src/tls.rs [hunk 3]
```diff
@@ -175,24 +172,60 @@ pub fn tls_config(config: &crate::config::Config) -> Result<Option<TlsData>, Err
...
+                        tracing::warn!(
+                            client_certs = ?clicerts,
+                            error = ?e,
+                            "Unable to load client certificates, continuing without them in TLS mode"
+                        );
+                    }
+                };
+            }
+            None => {
+                tracing::warn!(
```
Comment (nichtsfrei) [inline rust/openvasd/src/tls.rs:215]: shouldn't that be either a info or even debug as it is configured that way?

File: rust/openvasd/src/tls.rs [hunk 4]
```diff
@@ -175,24 +172,60 @@ pub fn tls_config(config: &crate::config::Config) -> Result<Option<TlsData>, Err
...
+                    "Client certificates option not set, continuing without them in TLS mode."
```
Comment (nichtsfrei) [inline rust/openvasd/src/tls.rs:216]: The wording is confusing. This is a normal TLS mode without client verification. So I would suggest to rephrase it to `Client verification disabled`?

File: rust/openvasd/src/tls.rs [hunk 5]
```diff
@@ -175,24 +172,60 @@ pub fn tls_config(config: &crate::config::Config) -> Result<Option<TlsData>, Err
...
-    config.alpn_protocols = vec![b"h2".to_vec()];
+        Ok(Some((client_identifier, config, !clients.is_empty())))
+    } else {
+        match &config.tls.client_certs {
+            Some(clicerts) => {
+                match std::fs::read_dir(clicerts) {
+                    Ok(_) => {
+                        tracing::warn!(
+                            client_certs = ?clicerts,
+                            "No valid client certificates found, continuing without them in TLS mode."
```
Comment (nichtsfrei) [inline rust/openvasd/src/tls.rs:202]: What's the benefit of `in TLS mode.`? Isn't it clear based on the protocol information on startup?

**Review by ArnoStiefvater — APPROVED** [review]

**Review by Kraemii — APPROVED** [review]
